targets/wasm_exec.js

```diff
@@ -298,11 +298,22 @@
 
 					// func finalizeRef(v ref)
 					"syscall/js.finalizeRef": (v_ref) => {
-						// Note: TinyGo does not support finalizers so this should never be
-						// called.
-						console.error('syscall/js.finalizeRef not implemented');
+						// Note: TinyGo does not support finalizers so this is only called
+						// for one specific case, by js.go:jsString. and can/might leak memory.
+						const id = mem().getUint32(unboxValue(v_ref), true);
+						if (this._goRefCounts?.[id] !== undefined) {
+							console.log("syscall/js.finalizeRef", id, this._goRefCounts[id]);
+							this._goRefCounts[id]--;
+							if (this._goRefCounts[id] === 0) {
+								const v = this._values[id];
+								this._values[id] = null;
+								this._ids.delete(v);
+								this._idPool.push(id);
+							}
+						} else {
+							console.log("syscall/js.finalizeRef: unknown id", id);
```

Comment:
Should probably remove this call to `log()` entirely, or else switch to `error()`. 

Assuming this entire PR is a good idea, which I don't have an opinion on at the moment. @ldemailly and @aykevl you both have a better grasp on the issues involved.

Reply:
It's definitely not an error() because that would break/defeat the fix and I think it helps showing what happens for future troubleshooting but if it's all that stands between merging I can also remove it

Reply:
It appears from looking at the big Go implementation and also the current TinyGo implementation, and also @aykevl comment, that they think the code change should probably be something like this:

```js
// func finalizeRef(v ref)
"syscall/js.finalizeRef": (v_ref) => {
        const id = mem().getUint32(unboxValue(v_ref), true);
	this._goRefCounts[id]--;
	if (this._goRefCounts[id] === 0) {
		const v = this._values[id];
		this._values[id] = null;
		this._ids.delete(v);
		this._idPool.push(id);
	}
}
```

Is that a correct summary?

Reply:
no, not sure how you get to that conclusion, sorry

https://github.com/tinygo-org/tinygo/pull/4352/files#r1688222655

doing that would just silently ignore the case where you get an unexpected id, which appears to be _all_ the cases and I don't think that's more helpful than a log

yet I suppose anything that doesn't involve `console.error` would be progress, not sure I get the resistance to this current version though.

Reply:
this being said... I thought console.error() was fatal (ie current version of the code) but it's not or I used a different setup when it hit me - so technically this is isn't needed at all, outside of generating console errors every invocation. I think this is progress but....

Reply:
https://github.com/golang/go/blob/master/misc/wasm/wasm_exec.js#L22

Reply:
as in when something is dubious a log is appropriate imo - but... will remove and nobody will know that it's non functional... but maybe we can move forward

Reply:
done

Reply:
> https://github.com/golang/go/blob/master/misc/wasm/wasm_exec.js#L22

The `wasm_exec.js` implementation for `finalizeRef` does not call `console.log` to be more specific:
https://github.com/golang/go/blob/master/misc/wasm/wasm_exec.js#L298-L309

Reply:
I am hiding the bug now,  even if I think it's wrong, can we merge?

Reply:
@ldemailly please see my other code comment, I think I understand better now what you intended.